platforms/ide/problems/src/main/kotlin/org/gradle/problems/internal/impl/DefaultProblemsReportCreator.kt

```diff
@@ -85,46 +86,69 @@ class DefaultProblemsReportCreator(
 
     override fun emit(problem: Problem, id: OperationIdentifier?) {
         problemCount.incrementAndGet()
-        report.onProblem(object : JsonSource {
-            override fun writeToJson(jsonWriter: JsonWriter) {
-                with(jsonWriter) {
-                    jsonObject {
-                        property("trace") {
-                            jsonObjectList(listOf(getFirstFileLocation())) { path ->
-                                property("kind", "Project")
-                                property("path", path)
+        report.onProblem(JsonProblemWriter(problem, failureDecorator, failureFactory))
+    }
+}
+
+class JsonProblemWriter(private val problem: Problem, private val failureDecorator: FailureDecorator, private val failureFactory: FailureFactory) : JsonSource {
+    override fun writeToJson(jsonWriter: JsonWriter) {
+        with(jsonWriter) {
+            jsonObject {
+                if (problem.locations.isNotEmpty()) {
```

Comment:
Shouldn't we filter for file locations here and check whether the result is empty?